yt_dlp/extractor/mlb.py

```diff
@@ -263,3 +267,73 @@ def _download_video_data(self, display_id):
   }
 }''' % display_id,
             })['data']['mediaPlayback'][0]
+
+
+class MLBTVIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?mlb\.com/tv/g(?P<id>\d{6})'
+    _NETRC_MACHINE = 'mlb'
+
+    _TESTS = [{
+        'url': 'https://www.mlb.com/tv/g661581/vee2eff5f-a7df-4c20-bdb4-7b926fa12638',
+        'info_dict': {
+            'id': '661581',
+            'ext': 'mp4',
+            'title': '2022-07-02 - St. Louis Cardinals @ Philadelphia Phillies',
+        },
+        'params': {
+            'skip_download': True,
+        },
+    }]
+
```

Comment:
All video on the site needs account, correct? Do this to give users a proper error message if try try without `-u`.
```suggestion
    _access_token = None
    
    def _real_initialize(self):
        if not self._access_token:
            self.raise_login_required(
                'All videos are only available to registered users', method='password')
```

Reply:
There are a small fraction of videos that are "free" (no paid subscription required) but I'm not sure if they still require an account or not. Either way, I am fine requiring an account for any video in the service.

Reply:
Just verified that a free video still requires an account